api-service/database/licenses_test.go

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) 2020 Sorint.lab S.p.A.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package database
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ercole-io/ercole/model"
+	"github.com/ercole-io/ercole/utils"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+func (m *MongodbSuite) TestSearchLicenses() {
+	defer m.db.Client.Database(m.dbname).Collection("hosts").DeleteMany(context.TODO(), bson.M{})
+	m.InsertHostData(utils.LoadFixtureMongoHostDataMap(m.T(), "../../fixture/test_apiservice_mongohostdata_17.json"))
+
+	defer m.db.Client.Database(m.dbname).Collection("licenses").DeleteMany(context.TODO(), bson.M{})
+	m.InsertLicense(model.LicenseCount{Name: "Oracle ENT"})
+	m.InsertLicense(model.LicenseCount{Name: "Diagnostics Pack"})
+	m.InsertLicense(model.LicenseCount{Name: "Real Application Clusters"})
```

Comment:
Maybe you should changes the test values so more cases are covered?

Reply:
(and the results variance are higher)

Reply:
Which cases are missing?

Reply:
License Count > 0
License Count == Used
In cluster/not in cluster
CostPerProcessor > 0
...

Reply:
.

Reply:
I don't added any more tests because goal of this PR was to add ListLicenses, not SearchLicenses, which I didn't wrote.
Let's close this PR to let frontend devs work and then I'll add more tests.